### DISCUSSION TITLE
How to implement ISelectionListener for an UIExtension

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1375] [Q&A]
Hi,
     I create an AbstractUIExtension implement ISelectionListener in glsp version 1.1.0
![image](https://github.com/user-attachments/assets/8cd58130-e667-4c7b-a6cc-38bc66acf484)
in that version the SelectionService has a register method.
![image](https://github.com/user-attachments/assets/e0f81fef-04df-4c2b-b910-7156234fc8c3)
but in glsp version 2.1.1. the register method in SelectionService  has been removed. the selectionChanged in my UIExtension can not be triggered when I select node. how can I trigger the selectionChanged in my UIExtension when I select item in my glsp language

**Comment by tortmayr** [comment]
Hi @wss29 
with GLSP 2.x the lister mechanism has been reworked to use a emitter/event system similar to how its used in Theia VSCode.
To manually register a listener for selection services you have to use the `SelectionService.onSelectionChanged()` method now.
```ts
  @postConstruct()
    protected postConstruct(): void {
        this.selectionService.onSelectionChanged(change =>
            this.onSelectionChanged(change => listener.selectionChanged(change.root, change.selectedElements, change.deselectedElements))
        );
    }
```

**Reply by wss29** [reply]
Hi @tortmayr 
   Does that mean I do not need to implement ISelectionListener for my UIExtension, directly use selectionService, and listen to the onSelectionChanged event in my UIExtension.
